pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -113,10 +113,15 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.ModelBuilde
 		RootVolumeType:         lc.RootVolumeType,
 		SSHKey:                 lc.SSHKey,
 		SecurityGroups:         lc.SecurityGroups,
-		SpotPrice:              lc.SpotPrice,
 		Tenancy:                lc.Tenancy,
 		UserData:               lc.UserData,
-	}, nil
+	}
+	// You cannot use a launch template that is set to request Spot Instances (InstanceMarketOptions)
```

Comment:
Should we output a warning in the logs if both MixedInstancesPolicy & SpotPrice are set informing the user that they will not be able to use spot instances?

Reply:
That error message occurs when the LaunchTemplate contains the spot price. This PR will fix it so that the spot price instead lives in the ASG's MixedInstancesPolicy, allowing both MixedInstancesPolicy and SpotPrice to work together. So I don't think theres anything to inform the user about, but I can add clarity to this comment.